src/mount_point.rs

```diff
@@ -0,0 +1,11 @@
+use std::path::{Path, PathBuf};
+
+pub fn find_mountpoint(path: &Path, mount_points: &[&Path]) -> Option<PathBuf> {
+    let path = path.to_string_lossy();
+
+    mount_points
+        .iter()
+        .filter(|mnt| path.starts_with(&*mnt.to_string_lossy()))
+        .max_by_key(|mnt| mnt.to_string_lossy().len())
+        .map(|mnt| mnt.to_path_buf())
+}
```

Comment:
1. Please rename it to `find_mount_point`.

2. Let's not expose this function as `pub`. As for the test, you can move it to this file.

```rust
#[cfg(test)]
mod tests {
    use super::find_mount_point;
    use std::path::Path;

   #[test]
   fn test_find_mount_point() { /* ... */ }
}
```

3. Doesn't `std::path::Path` already have a `start_with` method? Why convert it to UTF-8? As for `mnt.to_string_lossy().len()`, you can do `AsRef::<OsStr>::as_ref(mnt).len()`.

4. ~~Can you explain to me why use `max_by_key`?~~ It took me a while to understand. Can you please add this comment:

```rust
        .max_by_key(/* ... */) // mount points can be nested in each other
```

5. Finally, I don't think the final `.map(|mnt| mnt.to_path_buf())` is necessary. Just add a lifetime annotation and return `Option<&'a Path>` directly, like so:

```rust
pub(super) fn find_mount_point<'a>(
    path: &Path,
    mount_points: impl IntoIterator<Item = &'a Path>
) -> Option<&'a Path> { /* ... */ }
```